syntax = "proto3";

import "google/protobuf/any.proto";

option go_package = "/proto";


enum MessageType {
    GET_PTZDevice = 0;
    SET_MediaSettings = 1;
    GET_MediaSettings = 2;
    GET_ImageSettings = 3;
    SET_ImageSettings = 4;
    GET_VideoAdjustment = 5;
    SET_VideoAdjustment = 6;
    GET_ExposureSettings = 7;
    SET_ExposureSettings = 8;
    GET_DayNightSwitchSettings = 9;
    SET_DayNightSwitchSettings = 10;
    GET_WhiteBalanceSettings = 11;
    SET_WhiteBalanceSettings = 12;
    GET_InfraredImageSettings = 13;
    SET_InfraredImageSettings = 14;
    GET_OSDSettings = 15;
    SET_OSDSettings = 16;
    GET_IPSettings = 17;
    SET_IPSettings = 18;
    GET_PortSettings = 19;
    SET_PortSettings = 20;
    GET_FTPSettings = 21;
    SET_FTPSettings = 22;
    GET_PlatformSettings = 23;
    SET_PlatformSettings = 24;
    GET_AlarmSettings = 25;
    SET_AlarmSettings = 26;
    GET_GasLeakageAlarmSettings = 27;
    SET_GasLeakageAlarmSettings = 28;
    SET_UserAccountSettings = 29;
    GET_UserAccountSettings = 30;
    GET_LoginSecuritySettings = 31;
    SET_LoginSecuritySettings = 32;
    GET_TimeSettings = 33;
    SET_TimeSettings = 34;
    GET_SystemInfo = 35;
    GET_Reboot = 36;
    SET_Reboot = 37;

    QUERY_SYSTEM_LOGS = 38;

    SET_TrackingScanCmd = 39;
    SET_LinearScanSettings = 40;
    SET_LinearScanControl = 41;
    SET_Rotation = 42;
    SET_LensControl = 43;
    GET_LensControl = 44;
    GET_AccessoryControl = 45;
    SET_AccessoryControl = 46;
    SET_Scanning = 47;
}

message PTZControlProtol{
    MessageType type = 1;

    // 设备ID
    string device_id = 2;

    // 时间戳，表示消息发送或接收的时间
    uint64 timestamp = 3;
    oneof content {
        // 设备信息
        PTZDevice device = 4;

        // 设备控制参数
        MediaSettings media_settings = 5;

        ImageSettings image_settings = 6;

        ImageEnhancement image_enhancement = 7;

        VideoAdjustment video_adjustment = 8;

        ExposureSettings exposure_settings = 9;

        DayNightSwitchSettings day_night_switch_settings = 10;

        WhiteBalanceSettings white_balance_settings = 11;

        InfraredImageSettings infrared_image_settings = 12;

        OSDSettings osd_settings = 13;

        IPSettings ip_settings = 14;

        PortSettings port_settings = 15;

        FTPSettings ftp_settings = 16;

        PlatformSettings platform_settings = 17;

        AlarmSettings alarm_settings = 18;

        GasLeakageAlarmSettings gas_leakage_alarm_settings = 19;

        UserAccountSettings user_account_settings = 20;

        LoginSecuritySettings login_security_settings = 21;

        TimeSettings time_settings = 22;

        SystemInfo system_info = 23;

        Reboot reboot = 24;

        SystemLogsQuery system_logs_query = 25;

        LogEntry log_entry = 26;

        PresetPositionOp preset_position_op = 27;

        PresetList preset_list = 28;

        TrackingScan tracking_scan = 29;

        TrackingScanCmd tracking_scan_cmd = 30;

        LinearScanSettings linear_scan_settings = 31;

        LinearScanControl linear_scan_control = 32;

        Rotation rotation = 33;

        LensControl lens_control = 34;

        AccessoryControl accessory_control = 35;

        Scanning scanning = 36;

        ImageModeMessage image_mode_message = 37;
    }
}

message PTZDevice {
    // 设备名称，通常是设备的标识名称
    string name = 1; // 设备名称

    // 用户名，通常是与设备关联的用户名
    string username = 2; // 用户名

    // 端口号，设备的通信端口
    int32 port = 3; // 端口号

    // IP 地址，设备的网络地址
    string ip_address = 4; // IP 地址

    // 设备分组，设备所属的分组或类别
    string group = 5; // 设备分组

    // 设备状态，表示设备的在线或离线状态
    DeviceStatus status = 6; // 设备状态

    // 定义设备状态的枚举类型
    enum DeviceStatus {
        UNKNOWN = 0; // 异常
        ONLINE = 1;  // 在线
        OFFLINE = 2; // 离线
    }
}

// 定义视频和音频参数设置协议
message MediaSettings {
    // 码流类型，例如主码流或辅码流
    StreamType stream_type = 1; // 码流类型

    // 视频流类型，音视频混合
    VideoStreamType video_stream_type = 2; // 视频流类型

    // 视频分辨率，例如 1920x1080
    Resolution resolution = 3; // 分辨率

    // 码率类型，例如固定码率或可变码率
    BitrateType bitrate_type = 4; // 码率类型

    // 视频码流，例如 2048 kbps
    int32 bitrate = 5; // 码流256—4096Kbs可选

    // 视频帧率，例如 30 帧每秒
    FrameRate frame_rate = 6; // 视频帧率

    // 编码模式，例如实时编码或高质量编码
    VideoCodec encoding_mode = 7; // 编码模式

    // I 帧间隔，例如 60 帧
    int32 iframe_interval = 8; // I帧间隔最大为 150

    // 辅助码流开启状态
    bool auxiliary_stream_enabled = 9; // 辅助码流开启

    // 音频输入类型，例如内置麦克风或外部麦克风
    AudioSource audio_input = 11; // 音频输入

    // 音频采样频率，例如 44100 Hz
    int32 sampling_rate = 12; // 采样频率

    // 环境噪声过滤状态
    bool noise_reduction_enabled = 13; // 环境噪声过滤

    // 麦克风音量，例如 50%
    int32 microphone_volume = 14; // 麦克风音量

    // 扬声器音量，例如 75%
    int32 speaker_volume = 15; // 扬声器音量
}

// 图像调节设置
message ImageSettings {
    // 对比度调整
    int32 contrast = 1;

    // 亮度调整
    int32 brightness = 2;

    // 饱和度调整
    int32 saturation = 3;

    // 锐度调整
    int32 sharpness = 4;
    // 宽动态设置
    bool wide_dynamic_range = 5;

    // 可见光图像增强设置
    bool visible_light_enhancement = 6;

    // 抓图类型
    CaptureType capture_type = 7;
    enum CaptureType {
        SCHEDULED = 0; // 普通抓图
        TRIGGERED = 1; // 触发抓图
    }

    // 图片大小
    // ImageSize image_size = 8;

    // 图片质量
    ImageQuality image_quality = 9;
    enum ImageQuality {
        LOWEST = 0;
        LOWER = 1;
        LOW = 2;
        MEDIUM = 3;
        HIGHER = 4;
        HIGHEST = 5;
    }
}

// 图像增强设置
message ImageEnhancement {
    // 数字降噪设置
    NoiseReduction noise_reduction = 1; // 数字降噪功能设置

    // 雾透模式设置
    bool fog_enhancement = 2; // 雾透模式设置

    // 电子防抖设置
    bool stabilization = 3; // 电子防抖功能设置

    // 灰度范围设置
    int32 gray_range = 4; // 灰度范围值，范围是 [0-255]
}

// 数字降噪设置
message NoiseReduction {
    bool enabled = 1; // 是否开启数字降噪功能
    int32 level = 2;  // 降噪等级，开启时有效，范围是 [0-100]
}

// 视频调整参数
message VideoAdjustment {
    // 镜像设置
    MirrorSetting mirror_setting = 1; // 镜像模式选择

    // 电子防抖设置
    StabilizationSetting stabilization_setting = 2; // 电子防抖模式选择

    // 镜像设置
    enum MirrorSetting {
        NONE = 0; // 关闭镜像
        HORIZONTAL = 1; // 左右镜像
        VERTICAL = 2; // 上下镜像
    }

    // 电子防抖设置
    enum StabilizationSetting {
        OFF = 0; // 关闭电子防抖
        PAL = 1; // PAL 制（50Hz）
        NTSC = 2; // NTSC 制（60Hz）
    }
}

// 曝光设置
message ExposureSettings {
    // 曝光模式
    ExposureMode exposure_mode = 1;
    // 曝光模式的枚举定义
    enum ExposureMode {
        AUTO = 0;
        MANUAL = 1;
    }

    // 曝光时间（以毫秒为单位）
    int32 exposure_time = 2;
}

// 定时切换设置
message DayNightSwitchSettings {
    Mode mode = 1;
    enum Mode {
        AUTO = 0;
        DAY = 1;
        NIGHT = 2;
        TIMED = 3; // 定时切换模式
    }
    // 开始时间，24小时制
    string start_time = 2; // 格式例如 "08:00"

    // 结束时间，24小时制
    string end_time = 3; // 格式例如 "18:00"
}

// 白平衡设置
message WhiteBalanceSettings {
    WhiteBalanceMode white_balance_mode = 1;
    ManualWhiteBalance manual_white_balance = 2; // 可选
    enum WhiteBalanceMode {
        AUTO = 0;
        MANUAL = 1;
    }
    message ManualWhiteBalance {
        int32 red_gain = 1; // R 增益
        int32 blue_gain = 2; // B 增益
    }
}

// 红外光图像设置
message InfraredImageSettings {
    // 图像调节参数
    int32 brightness = 1; // 亮度调整，范围是 [0-100]
    int32 contrast = 2;   // 对比度调整，范围是 [0-100]
    int32 denoise = 3;    // 去噪调整，范围是 [0-100]

    // DDE模式设置
    int32 intensity = 4;  // DDE强度调整，范围是 [0-100]
    bool detail_enhancement = 5; // 是否开启细节增强效果
    bool edge_enhancement = 6;   // 是否开启边缘增强效果
    bool stripe_filter = 7;      // 是否开启条纹滤波功能
}

// OSD设置
message OSDSettings {
    // 通道标题设置
    bool enabled = 1; // 是否开启通道标题显示
    string name = 2;  // 通道标题名称

    // 时间标题设置
    bool time_enabled = 3;         // 是否开启时间标题显示
    bool show_weekday = 4;    // 是否开启星期显示
    TimeFormat time_format = 5; // 时间制式（24小时制或12小时制）
    DateFormat date_format = 6; // 日期格式

    // 云台信息设置
    bool preset_enabled = 7;    // 是否显示预置位
    bool zoom_enabled = 8;      // 是否显示放大倍数
    bool cruise_enabled = 9;    // 是否显示巡航
    bool tracking_enabled = 10;  // 是否显示寻迹
    bool scan_enabled = 11;      // 是否显示线性扫描

    // 字符叠加设置
    Size size = 12;       // 字符大小
    Color color = 13;     // 字符颜色
    Alignment alignment = 14; // 字符对齐方式
    enum Size {
        LARGE = 0;  // 大
        MEDIUM = 1; // 中（默认）
        SMALL = 2;  // 小
    }

    enum Color {
        AUTO = 0;   // 黑白自适应
        WHITE = 1;  // 白色
        BLACK = 2;  // 黑色
    }

    enum Alignment {
        LEFT = 0;   // 左对齐
        CENTER = 1; // 居中
        RIGHT = 2;  // 右对齐
    }
    // 时间制式
    enum TimeFormat {
        TWENTY_FOUR_HOUR = 0; // 24小时制
        TWELVE_HOUR = 1;      // 12小时制
    }

    // 日期格式
    enum DateFormat {
        YEAR_MONTH_DAY = 0; // 年-月-日
        MONTH_DAY_YEAR = 1; // 月-日-年
        DAY_MONTH_YEAR = 2; // 日-月-年
    }
}

message IPSettings {
    bool is_static_ip = 1;
    // IPv4 地址设置
    message IPv4Config {
        string ip_address = 1;      // IPv4 地址
        string subnet_mask = 2;     // 子网掩码
        string default_gateway = 3; // 默认网关
        string primary_dns = 4;   // 首选 DNS 服务器 IP 地址
        string secondary_dns = 5; // 备用 DNS 服务器 IP 地址
    }

    // IPv6 地址设置
    message IPv6Config {
        string ip_address = 1;      // IPv6 地址
        string subnet_prefix = 2;   // 子网前缀
        string default_gateway = 3; // 默认网关
        string primary_dns = 4;   // 首选 DNS 服务器 IP 地址
        string secondary_dns = 5; // 备用 DNS 服务器 IP 地址
    }
}

// 设备端口设置
message PortSettings {
    // 最大连接数
    string max_connections = 1; // 可设置设备的最大连接数

    // 监听端口
    string listen_port = 2; // 设备用于监听的端口

    // 视频数据发送端口
    string video_data_port = 3; // 用于发送视频数据的端口

    // HTTP 端口
    string http_port = 4; // 用于 HTTP 请求的端口

    // RTSP 端口
    string rtsp_port = 5; // 用于 RTSP 流的端口

    // HTTPS 端口
    string https_port = 6; // 用于 HTTPS 请求的端口
}

// FTP 服务设置
message FTPSettings {
    // FTP 服务的 IP 地址
    string ftp_ip = 1; // FTP 服务的 IP 地址

    // 存储路径
    string storage_path = 2; // FTP 服务存储文件的路径

    // 端口
    string port = 3; // FTP 服务监听的端口

    // 账号
    string username = 4; // 用于访问 FTP 服务的账号

    // 密码
    string password = 5; // 用于访问 FTP 服务的密码
}

// 平台接入设置
message PlatformSettings {
    // 平台的 IP 地址
    string platform_ip = 1; // 平台 IP 地址，用于连接到平台

    // 平台的端口
    string platform_port = 2; // 平台服务监听的端口

    // 接入协议类型
    ProtocolType protocol = 3; // 选择的接入协议类型（GB28181、ONVIF 或其他）
    // 接入协议类型
    enum ProtocolType {
        PROTOCOL_UNKNOWN = 0; // 未知协议
        GB28181 = 1; // GB28181 协议
        ONVIF = 2; // ONVIF 协议
        OTHER = 3; // 其他平台协议
    }

    // API key
    string key = 4; // 用于平台接入的 API key

    // API secret
    string secret = 5; // 用于平台接入的 API secret
}

// 移动侦测设置
message AlarmSettings {
    // 移动侦测启用状态
    bool enabled = 1; // 启用或关闭移动侦测

    // 移动侦测区域设置
    repeated DetectionArea areas = 2; // 设置多个移动侦测区域

    // 布放时间
    repeated string setup_time = 3; // 移动侦测区域布放的时间（以秒为单位）

    // 报警联动设置
    // 是否录像
    bool record_video = 4; // 发生报警时是否进行录像
  // 录像延时时间
    int32 record_delay = 5; // 录像延时时间，单位秒（0—300秒）

    // 是否报警输出
    bool alarm_output = 6; // 发生报警时是否进行报警输出
  // 报警输出延时时间
    int32 output_delay = 7; // 报警输出延时时间，单位秒（0—300秒）

    // 是否发送邮件
    bool send_email = 8; // 发生报警时是否发送邮件
  // 邮件接收地址
    string email_address = 9; // 发送报警信息的目标邮箱地址

    // 是否抓图
    bool snapshot = 10; //发生报警时是否抓图
  // 抓图时间间隔
    int32 snapshot_interval = 11; // 抓图时间间隔，单位秒（10—3600秒）

    // 定义移动侦测区域
    repeated DetectionArea detection_areas = 12;

    AlarmType alarm_type = 13;
    enum AlarmType {
        MOTION = 0;
        COVER = 1;
    }
}

// 定义移动侦测区域
message DetectionArea {
}

message GasLeakageAlarmSettings {
    bool alarm_enable = 1;
    // 报警浓度值
    float alarm_concentration = 2; // 触发报警的气体浓度值（5—50000ppm.m）

    // 是否自动拍照
    bool auto_snapshot = 3; // 报警时是否自动拍照
  // 抓图时间间隔
    int32 snapshot_interval = 4; // 抓图时间间隔，单位秒（10—3600秒）

    // 是否自动录制视频
    bool auto_record_video = 5; // 报警时是否自动录制视频
  // 录像延时时间
    int32 record_delay = 6; // 录像延时时间，单位秒（0—300秒）

    // 是否输出报警信号
    bool alarm_output = 7; // 报警时是否进行报警输出
    int32 alarm_delay = 8; // 报警延时

    // 是否发送邮件
    bool send_email = 9; // 报警时是否发送邮件
  // 邮件接收地址
    string email_address = 10; // 发送报警信息的目标邮箱地址
}

// 用户账户设置
message UserAccountSettings {
    // 操作请求
    string request_id = 1;
    string request_pwd = 2;

    OperationType operation_type = 3; // 操作类型
    string target_user_id = 4; // 目标用户ID
    string new_password = 5; // 新密码（如果操作类型是修改密码时需要）

    // 操作类型定义
    enum OperationType {
        OPERATION_UNSPECIFIED = 0; // 未指定
        ADD = 1; // 新增
        DELETE = 2; // 删除
        MODIFY = 3; // 修改
        SUSPEND = 4; // 停用
        RESUME = 5; // 启用
    }
}

// 登录安全设置
message LoginSecuritySettings {
    int32 max_failed_attempts = 1; // 密码容错次数，范围：0-99，默认为5次
    int32 lockout_duration = 2; // 锁定时间（分钟），范围：1-999分钟，默认为5分钟
}

// 时间设置
message TimeSettings {
    // 时区设置
    int32 time_zone = 1;
    bool auto_sync_local_time = 2; // 是否自动同步PC本地时间

    bool enable_ntp_sync = 3; // 是否启用平台自动校时（NTP）
    string ntp_server = 4; // NTP服务器地址
    int32 sync_interval = 5; // 同步周期（分钟）

    // 手动校时设置
    string manual_time = 6;
}

// 系统信息
message SystemInfo {
    // 设备名称
    string device_name = 1; // 设备名称，例如 "SensorX"

    // 设备编号
//    string device_id = 2; // 设备唯一标识符，例如 "1234567890"

    // 主控版本
    string main_controller_version = 3; // 主控系统的版本号，例如 "v1.2.3"

    // 算法版本
    string algorithm_version = 4; // 使用的算法版本，例如 "alg-v2.1.0"

    // 模型版本
    string model_version = 5; // 当前使用的模型版本，例如 "model-2024.09"
}

message Reboot {
    // 立即重启
    bool reboot_now = 1;
    // 开启/关闭定制重启功能
    bool enable_custom_reboot = 2; // true 表示开启定制重启，false 表示关闭

    // 设置定制重启的时间点
    // 每周的某一天和时间点，定制重启将在设置的时间点自动重启设备
    repeated WeeklyRebootSchedule schedules = 3;

    // 定义每周的定制重启时间点
    message WeeklyRebootSchedule {
        // 星期几（0 = 周日, 1 = 周一, ..., 6 = 周六）
        int32 day_of_week = 1;

        // 时间点，使用24小时制表示（例如 "14:30" 表示下午2点30分）
        string time_of_day = 2;
    }
    //恢复出厂
    bool reset = 4;
}

// 系统日志记录配置
message SystemLogsQuery {
    QueryType type = 1;
    string query_pattern = 2;
    // 日志列表
    repeated LogEntry log_entries = 3;

    enum QueryType {
        USER_ID = 0;
        LOG_TYPE = 1;
        OP_TYPE = 2;
        OP_DATE = 3;
    }

    // 时间戳，表示消息发送或接收的时间
    uint64 time = 4;
}

// 日志条目
message LogEntry {
    // 日志类型
    LogType log_type = 1;

    // 操作账户
    AccountType account_type = 2;

    // 操作日期（精确到秒）
    string operation_datetime = 3;

    // 请求IP
    string request_ip = 4;

    // 操作记录
    OperationType operation_type = 5;

    // 操作结果
    OperationResult operation_result = 6;

    // 日志类型
    enum LogType {
        OPERATION_LOG = 0; // 操作日志
        QUERY_LOG = 1;     // 查询日志
    }

    // 操作账户类型
    enum AccountType {
        SUPER_ADMIN = 0; // 超级管理员账户
        ADMIN = 1;       // 管理员账户
        CUSTOMER = 2;    // 客户账户
    }

    // 操作结果
    enum OperationResult {
        SUCCESS = 0; // 操作成功
        FAILURE = 1; // 操作失败
    }

    // 操作记录类型
    enum OperationType {
        INFO_QUERY = 0;    // 信息查询
        PARAM_CONFIG = 1;  // 参数配置
        ACCOUNT_MODIFY = 2; // 账号修改
        ACCOUNT_DELETE = 3; // 账号删除
        ACCOUNT_ADD = 4;   // 新增账号
        FACTORY_RESET = 5; // 恢复出厂设置
        SYSTEM_UPGRADE = 6; // 系统升级
    }
}

message PresetPositionOp {
    enum PresetOperatType {
        ADD = 0;
        DEL = 1;
        MODIFY = 2;
    }
    string target = 1;
}
// 定义包含预置位列表的消息
message PresetList {
    // 预置位列表
    repeated PresetPosition preset_positions = 1;
}

// 巡航组的消息定义
message CruiseGroup {
    // 巡航组的名称
    string name = 1; // 如 "Group1", "Group2" 等

    // 预置位列表
    repeated PresetPosition preset_positions = 2; // 每个巡航组最多 20 个预置位点
}

// 定义预置位的消息
message PresetPosition {
    // 预置位的名称
    string name = 1; // 如 "Preset1", "Preset2" 等

    // 预置位的编号，递增数字
    int32 number = 2; // 递增的数字，例如 1, 2, 3 等

    // 停留时间（以秒为单位）
    int32 dwell_time_seconds = 3; // 预置位点的停留时间
}

message TrackingScan {
    repeated TrackingScanCmd cmd = 1; //最大5条
}

message TrackingScanCmd {
    OperationType type = 1;
    // 操作类型的枚举
    enum OperationType {
        PAN_TILT = 0; // 水平/垂直运动
        ZOOM = 1; // 变倍
        PRESET_CALL = 2; // 预置点调用
    }

    // 水平运动的角度变化
    float pan_angle = 2; // 角度，单位为度
  // 垂直运动的角度变化
    float tilt_angle = 3; // 角度，单位为度
  // 变倍的倍数变化
    float zoom_factor = 4; // 变倍系数，例如 1.0 表示原始倍数
  // 预置点的编号
    int32 preset_number = 5; // 调用的预置点编号
}

// 线性扫描设置
message LinearScanSettings {
    int32 scan_id = 1; // 线性扫描编号 (1-8)
    int32 scan_speed = 2; // 扫描速度 (1-10)
    int32 left_pause_time = 3; // 左侧停留时间 (0-600秒)
    int32 right_pause_time = 4; // 右侧停留时间 (0-600秒)
    int32 left_boundary = 5; // 左边界坐标
    int32 right_boundary = 6; // 右边界坐标
}

// 开始或停止线性扫描
message LinearScanControl {
    int32 scan_id = 1; // 线性扫描编号 (1-8)
    bool start = 2; // 启动扫描 (true) 或停止扫描 (false)
}

enum ControlType {
    UNKNOWN = 0;
    ROTATION = 1;
    LENS_CONTROL = 2;
    ACCESSORY_CONTROL = 3;
    PRESET_POSITION = 4;
    SCANNING = 5;
}
// 旋转方向
enum Direction {
    UP = 0;
    DOWN = 1;
    LEFT = 2;
    RIGHT = 3;
}

// 云台旋转指令
message Rotation {
    Direction direction = 1; // 旋转方向
    float speed = 2;         // 旋转速度 (0-1 范围)
}

// 云台变倍、变焦及光圈控制
message LensControl {
    float zoom_level = 1;   // 变倍级别
    float focus = 2;        // 焦距
    float aperture = 3;     // 光圈
}

// 补光灯和雨刷控制
message AccessoryControl {
    bool light = 1;         // 补光灯开关
    bool wiper = 2;         // 雨刷开关
}

// 扫描功能
message Scanning {
    enum ScanType {
        UNKNOWN = 0;
        CONTINUOUS = 1; // 续航扫描
        TRACKING = 2;   // 寻迹扫描
        LINEAR = 3;     // 线性扫描
    }
    ScanType scan_type = 1;  // 扫描类型
}

// 定义消息类型，用于包含图像模式数据
message ImageModeMessage {
    // 图像模式字段，使用枚举类型
    ImageMode mode = 1;
    // 定义枚举类型，用于表示不同的图像模式
    enum ImageMode {
        // 气云模式
        AIR_MEASUREMENT_CLOUD = 0;
        // TDLAS 模式
        TDLAS_CLOUD_CONCENTRATION = 1;
        // 红外普通模式
        INFRARED_NORMAL = 2;
        // 红外 PIP 模式
        INFRARED_PIP = 3;
        // 红外增强模式
        INFRARED_ENHANCED = 4;
        // 红外融合模式
        INFRARED_FUSION = 5;
        // 伪彩色白热模式
        PSEUDO_COLOR_WHITE_HOT = 6;
        // 伪彩色黑热模式
        PSEUDO_COLOR_BLACK_HOT = 7;
        // 伪彩色铁红模式
        PSEUDO_COLOR_IRON_RED = 8;
        // 伪彩色青色模式
        PSEUDO_COLOR_CYAN = 9;
        // 伪彩色彩虹模式
        PSEUDO_COLOR_RAINBOW = 10;
        // 伪彩色逆彩虹模式
        PSEUDO_COLOR_INVERSE_RAINBOW = 11;
        // 伪彩色红棕模式
        PSEUDO_COLOR_RED_BROWN = 12;
        // 伪彩色热铁模式
        PSEUDO_COLOR_HEAT_IRON = 13;
        // 伪彩色冷模式
        PSEUDO_COLOR_COLD = 14;
        // 伪彩色火焰模式
        PSEUDO_COLOR_FIRE = 15;
    }
}

// 码流类型的枚举定义
enum StreamType {
    MAIN = 0; // 主码流
    AUXILIARY = 1; // 辅码流
    THIRD = 2;
    FOURTH = 3;
    FIFTH = 4;
}

// 视频流类型的枚举定义
enum VideoStreamType {
    VIDEO = 0; // H.264 编码
    VIDEO_AUDIO = 1; // H.265 编码
}

// 分辨率的枚举定义
enum Resolution {
    RESOLUTION_2560x1440 = 0; // 2560x1440
    RESOLUTION_1920x1080 = 1; // 1920x1080
    RESOLUTION_1280x960 = 2;  // 1280x960
    RESOLUTION_1280x720 = 3;  // 1280x720
    RESOLUTION_704x576 = 4;   // 704x576
    RESOLUTION_640x480 = 5;   // 640x480
}

// 码率类型的枚举定义
enum BitrateType {
    CONSTANT = 0; // 固定码率
    VARIABLE = 1; // 可变码率
}

enum AudioSource {
    LineIn = 0;
    MicIn = 1;
}

// 视频编码模式的枚举定义
enum VideoCodec {
    H264 = 0;
    H265 = 1;
    MPEG4 = 2;
    MJPEG = 3;
}

// 音频编码模式的枚举定义
enum AudioCodec {
    G722 = 0; // 默认编码模式
    G711 = 1;
    ACC = 2;
}

// 帧率的枚举定义
enum FrameRate {
    FPS_10 = 0;
    FPS_15 = 1;
    FPS_20 = 2;
    FPS_25 = 3;
    FPS_30 = 4;
    FPS_35 = 5;
    FPS_40 = 6;
    FPS_45 = 7;
    FPS_50 = 8;
    FPS_55 = 9;
    FPS_60 = 10;
}